lib/radiator_web/controllers/api/audio_controller.ex

```diff
@@ -0,0 +1,59 @@
+defmodule RadiatorWeb.Api.AudioController do
+  use RadiatorWeb, :controller
+  use Radiator.Constants
+
+  alias Radiator.Directory.Editor
+
+  action_fallback RadiatorWeb.Api.FallbackController
+
+  def create(conn, %{"audio" => params = %{"network_id" => network_id}}) do
+    with user = current_user(conn),
+         {:ok, network} <- Editor.get_network(user, network_id),
+         {:ok, audio} <- Editor.create_audio(user, network, params) do
+      conn
+      |> put_status(:created)
+      |> put_resp_header("location", Routes.api_audio_path(conn, :show, audio))
+      |> render("show.json", %{audio: audio})
+    end
+  end
+
+  def create(conn, %{"audio" => params = %{"episode_id" => episode_id}}) do
+    with user = current_user(conn),
+         {:ok, episode} <- Editor.get_episode(user, episode_id),
+         {:ok, audio} <- Editor.create_audio(user, episode, params) do
+      conn
+      |> put_status(:created)
+      |> put_resp_header("location", Routes.api_audio_path(conn, :show, audio))
+      |> render("show.json", %{audio: audio})
+    end
+  end
+
+  def create(conn, %{"audio" => _}) do
+    conn
+    |> put_status(422)
+    |> json(%{"errors" => %{"assignment" => "Either episode_id or network_id must be present."}})
+  end
+
+  def show(conn, %{"id" => id}) do
+    with user = current_user(conn),
+         {:ok, audio} <- Editor.get_audio(user, id) do
+      render(conn, "show.json", %{audio: audio})
+    end
+  end
+
+  def update(conn, %{"id" => id, "audio" => audio_params}) do
+    with user = current_user(conn),
+         {:ok, audio} <- Editor.get_audio(user, id),
+         {:ok, audio} <- Editor.update_audio(user, audio, audio_params) do
+      render(conn, "show.json", %{audio: audio})
+    end
+  end
+
+  def delete(conn, %{"id" => id}) do
+    with user <- current_user(conn),
+         {:ok, audio} <- Editor.get_audio(user, id),
+         {:ok, _} <- Editor.delete_audio(user, audio) do
+      send_resp(conn, 204, "")
```

Comment:
isn't the convention to return the deleted entity?

Reply:
Not sure, couldn't find a definitive answer. The most "proper" excerpt I can offer is from https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.7 about the DELETE verb in general:

> A successful response SHOULD be 200 (OK) if the response includes an entity describing the status, 202 (Accepted) if the action has not yet been enacted, or 204 (No Content) if the action has been enacted but the response does not include an entity. 

I also found arguments that DELETE should return a 20x instead of 404 for nonexistent entities to make it idempotent. Then 204 is the only option.

In any case, we should write down how we treat DELETE in general and put it in the docs. Not sure if it's useful for UIs to get the deleted object back in the response (the UI should already have the object?) but the main concern is consistency.